### PR TITLE
[Python Tests] Refactoring of BlockchainHelper

### PR DESCRIPTION
**Pull request by ptagl**
This PR includes a small refactoring of the BlockchainHelper class, in particular of the `create_sidechain()` method, just to remove the explicit `version` input parameter that was redundant as already available in the `creation_arguments` input object.

Additional changes:
- Fixed an issue in `_generate_params()` (`mc_test.py`), which was returning `None` when called more than once with the same arguments
- Added two missing Python tests to the CI

### CODE REVIEW

File: qa/rpc-tests/test_framework/blockchainhelper.py [hunk 1]
```diff
@@ -165,15 +179,19 @@ def init_proof_keys(self, bitcoin_test_framework):
...
+        Creates a sidechain with the given name
+        (used to store it locally in a map) and
+        the creation arguments provided.
+        Note that among these arguments, only
+        "version" is mandatory.
         """
 
-        sc_input = SidechainCreationInput.from_rpc_args(self, sc_name, sidechain_version, optional_parameters)
+        # "version" is a mandatory argument
+        assert("version" in creation_arguments)
```
Comment (a-petrini) [inline qa/rpc-tests/test_framework/blockchainhelper.py]: This check is redundant, as it is performed also inside `SidechainCreationInput.from_rpc_args()`

File: qa/rpc-tests/test_framework/blockchainhelper.py [hunk 2]
```diff
@@ -27,6 +27,16 @@
...
 MODULUS_BITS = 255
 
+"""
+Dictionary of default parameters that can be used when calling 'create_sidechain'.
+"""
+SidechainParameters = {
+    "DEFAULT_SC_V0":                { "version": 0 },
+    "DEFAULT_SC_V1":                { "version": 1 },
+    "DEFAULT_SC_V2_CEASABLE":       { "version": 2, "withdrawalEpochLength": 10 },
+    "DEFAULT_SC_V2_NON_CEASABLE":   { "version": 2, "withdrawalEpochLength": 0 }
```
Comment (JackPiri) [inline qa/rpc-tests/test_framework/blockchainhelper.py]: Could be worth to add `"wCeasedVk": ""` for non-ceasable sidechains?

**Review by JackPiri — APPROVED** [review]
The PR is fine to me

**Review by a-petrini — CHANGES_REQUESTED** [review]

**Review by drgora — APPROVED** [review]
Looks good to me

File: qa/rpc-tests/test_framework/blockchainhelper.py (hunk 1 above)
Comment (a-petrini) [reply qa/rpc-tests/test_framework/blockchainhelper.py]: Fix looks good to me!

**Review by a-petrini — APPROVED** [review]

File: qa/rpc-tests/test_framework/blockchainhelper.py (hunk 2 above)
Comment (ptagl) [reply qa/rpc-tests/test_framework/blockchainhelper.py]: I like this suggestion as it also avoids having an ugly check inside `from_rpc_args()` to eventually unset `wCeasedVk`.
Fixed in the latest commit, thanks!

File: qa/rpc-tests/test_framework/blockchainhelper.py (hunk 1 above)
Comment (ptagl) [reply qa/rpc-tests/test_framework/blockchainhelper.py]: I agree, thanks for spotting, fixed in the latest commit.

**Pull request merged**